protocols/v2/roles-logic-sv2/src/channel_logic/channel_factory.rs

```diff
@@ -409,28 +413,28 @@ impl ChannelFactory {
             // If we have just a prev hash we need to send it after the SetupConnectionSuccess
             // message
             (Some((prev_h, _)), None, true) => {
-                let prev_h = prev_h.into_set_p_hash(channel_id);
+                let prev_h = prev_h.into_set_p_hash(channel_id, None);
                 result.push(Mining::SetNewPrevHash(prev_h.clone()));
                 Ok(())
             }
             // If we have a prev hash and a last valid job we need to send before the prev hash and
             // the the valid job
             (Some((prev_h, _)), Some(job), true) => {
-                let prev_h = prev_h.into_set_p_hash(channel_id);
-                result.push(Mining::SetNewPrevHash(prev_h.clone()));
+                let prev_h = prev_h.into_set_p_hash(channel_id, Some(job.job_id));
                 result.push(Mining::NewMiningJob(job));
```

Comment:
make sure future_job is set to true and NPH job id is the one of the job

Reply:
So the code above is the thing that makes sure the SNPH and NMJ job_ids match. future_job is set true otherwise this case wouldn't match (Some(job)) comes from the `last_valid_job` which originates in the `future_jobs` vec.

Reply:
`last_valid_job` can originate from a future job/template or from a non future job/template and is always non future per definition. A valid job is a job that need to be mined so not future.

Reply:
done